src/formatDate.js

```diff
@@ -9,6 +9,46 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const fromDelimeter = fromFormat[fromFormat.length - 1];
+  const dateArr = date.split(fromDelimeter);
+  const dataObj = {};
+
+  for (let d = 0; d <= 2; d++) {
+    dataObj[fromFormat[d]] = dateArr[d];
+  }
+
+  const properObj = {};
+  const propFormat = toFormat.slice(0, -1);
+
+  for (const d of propFormat) {
+    if (d === 'YY' || d === 'YYYY') {
+      properObj[d] = dataObj['YY'] || dataObj['YYYY'];
+    } else {
+      properObj[d] = dataObj[d];
+    }
+  }
+
+  if (('' + properObj['YY']).length === 4) {
+    properObj['YY'] = ('' + properObj['YY']).slice(2);
+  }
+
+  if (('' + properObj['YYYY']).length === 2) {
+    if (properObj['YYYY'] < 30) {
+      properObj['YYYY'] = '20' + properObj['YYYY'];
+    } else if (properObj['YYYY'] >= 30) {
+      properObj['YYYY'] = '19' + properObj['YYYY'];
+    }
```

Comment:
The condition and conversion logic here assume that the 'YYYY' property will always be two digits long when in fact 'YYYY' should always be four digits. It looks like you've swapped the 'YY' and 'YYYY' logic. You need to correct this so that 'YY' is expanded to 'YYYY' when necessary.